ctrd/container.go

```diff
@@ -562,47 +562,22 @@ func (c *Client) KillContainer(ctx context.Context, id string, sig int) (*Messag
 		return nil, fmt.Errorf("failed to send signal %d to container %s: %v", sig, id, err)
 	}
 
-	// if container task has been stopped, we should wait container and task to be totally deleted
-	time.Sleep(10 * time.Millisecond)
-
-	lc, err := wrapperCli.client.LoadContainer(ctx, id)
-	if err != nil {
-		return nil, err
+	var msg *Message
+	waitExit := func() *Message {
+		// FIXME: This wait for timeout process consumes too much time
+		return c.ProbeContainer(ctx, id, time.Duration(100)*time.Microsecond)
 	}
 
-	var (
-		status  containerd.Status
-		deleted bool
-	)
-
-	task, err := lc.Task(ctx, nil)
-	// the container task has been deleted
+	msg = waitExit()
+	err = msg.RawError()
 	if err != nil {
-		if errdefs.IsNotFound(err) {
-			deleted = true
-		} else {
-			return nil, err
-		}
-	} else {
-		// the container task isn't deleted, we could get task's status
-		status, err = task.Status(ctx)
-		if err != nil {
-			return nil, err
+		// the container task is still running after receiving client signal
+		if errtypes.IsTimeout(err) {
+			return nil, nil
```

Comment:
timeout should also return error

Reply:
@HusterWan No, I think we should just catch this timeout error and do nothing, the timeout error occurs just because this client signal doean't kill container(task).

Reply:
how do you distinguish signals that kill or not kill task ?

if the signal is `15`, but the task not exited before timeout , this case is normal ?

Reply:
Yes, you are right.
1. I just distinguish different signals according to task's status after receiving client singal. 
2. This case is abnormal, so we should set a proper timeout value......

Reply:
I think the main problem of this is that the containerd task Kill API is **asynchronous**. If the `pack.task.Kill()` process hasn't been finished, the program will still run to next code.

Reply:
And if we want to make this process **synchronous**, we could set a wait channel in `pack.task.Kill()` process to make sure the container task has been successfully killed, just like this:
```
	s, err := pack.task.Wait(ctx)
	if err != nil {
		return nil, err
	}

	// send signal to container
	if err := pack.task.Kill(ctx, syscall.Signal(sig), containerd.WithKillAll); err != nil {
		return nil, fmt.Errorf("failed to send signal %d to container %s: %v", sig, id, err)
	}
	<-s
```
But another problem is that if client signal is harmless(such as `SIGWINCH`, etc), this wait channel will be blocked forever...